apps/prairielearn/src/middlewares/authzCourseOrInstance.ts

```diff
@@ -53,13 +54,67 @@ const SelectAuthzDataSchema = z.object({
   permissions_course_instance: SprocAuthzCourseInstanceSchema,
 });
 
+type SelectAuthzData = z.infer<typeof SelectAuthzDataSchema>;
+
 const SelectUserSchema = z.object({
   user: UserSchema,
   institution: InstitutionSchema,
   is_administrator: z.boolean(),
   is_instructor: z.boolean(),
 });
 
+// This is if isCourseInstance is false
```

Comment:
But also when it's true, right? Given that `ResLocalsCourseInstanceAuthz` extends this?


```suggestion
```